### PR TITLE
common Update docker task to run instrumented tests and agent

### PR DESCRIPTION
**Pull request by p3dr0rv**
- Make the template  more flexible
- Add parameters for  the 
  - Azure Container Registry Image 
  - repository to checkout 
  -  Environment VSTS Maven Account 
- Download Image is not installed in Agent
-  Update Docker Build Agent 
   - Custom self-hosted AzureDevOps pools have the potential to be a surface area of attack for malicious actors.
To reduce the risk associated with custom pools we will need to use 1ES Hosted Pools.
   - https://www.1eswiki.com/wiki/1ES_hosted_AzureDevOps_Agents

### CODE REVIEW

File: azure-pipelines/templates/steps/auth-client/docker-tasks.yml [hunk 1]
```diff
@@ -1,12 +1,36 @@
-# File: docker-tasks.yml
+# File: azure-pipelines\templates\docker-instrumented-test.yml
+# Description: Template to run Instrumented test for Supported API levels
+
+parameters:
+- name: acrImage
```
Comment (iamgusain) [inline azure-pipelines/templates/steps/auth-client/docker-tasks.yml:5]: >acrImage [](http://example.com/codeflow?start=8&length=8)

nit: azureContainerRegistryImage

**Review by iamgusain — APPROVED** [review]
Approve with suggestion

File: azure-pipelines/templates/steps/auth-client/docker-tasks.yml (hunk 1 above)
Comment (p3dr0rv) [reply azure-pipelines/templates/steps/auth-client/docker-tasks.yml]: 1d2a95c2e43cffcc12daf7027ac04af9b66099bd

**Review by shoatman — APPROVED** [review]

File: azure-pipelines/templates/steps/auth-client/docker-tasks.yml [hunk 2]
```diff
@@ -1,22 +1,47 @@
...
   condition: always()
-  displayName: 'Post build clean up'
-- checkout: self
+  displayName: Post build clean up
+- checkout: ${{ parameters.androidProject }}
   clean: true
   submodules: recursive
+- script: touch env.list
+  displayName: 'Create Enviroment file'
+- script: echo "${{ parameters.EnvVstsMvnAccount }}=$(mvnAccessToken)" >> env.list
```
Comment (AdamBJohnsonx) [inline azure-pipelines/templates/steps/auth-client/docker-tasks.yml:21]: Are we trying to append to the file in this case?

**Review by AdamBJohnsonx — APPROVED** [review]

File: azure-pipelines/templates/steps/auth-client/docker-tasks.yml (hunk 2 above)
Comment (p3dr0rv) [reply azure-pipelines/templates/steps/auth-client/docker-tasks.yml:21]: yes

**Pull request merged**